onedal/datatypes/_data_conversion.py

```diff
@@ -22,37 +22,71 @@
 from onedal import _backend, _is_dpc_backend
 
 from ..utils import _is_csr
-from ..utils._dpep_helpers import is_dpctl_available
-
-dpctl_available = is_dpctl_available("0.14")
+from ..utils._dpep_helpers import dpctl_available, dpnp_available
 
 if dpctl_available:
-    import dpctl
     import dpctl.tensor as dpt
 
 
-def _apply_and_pass(func, *args):
+def _apply_and_pass(func, *args, **kwargs):
     if len(args) == 1:
-        return func(args[0])
-    return tuple(map(func, args))
-
-
-def from_table(*args):
-    return _apply_and_pass(_backend.from_table, *args)
-
-
-def convert_one_to_table(arg):
-    if dpctl_available:
-        if isinstance(arg, dpt.usm_ndarray):
-            return _backend.dpctl_to_table(arg)
+        return func(args[0], **kwargs)
+    return tuple(map(lambda arg: func(arg, **kwargs), args))
+
+
+def convert_one_from_table(table, sycl_queue=None, sua_iface=None, xp=None):
+    # Currently only `__sycl_usm_array_interface__` protocol used to
+    # convert into dpnp/dpctl tensors.
+    if sua_iface:
+        if _is_dpc_backend:
+            if (
+                sycl_queue
+                and sycl_queue.sycl_device.is_cpu
+                and table.__sycl_usm_array_interface__["syclobj"] is None
+            ):
+                # oneDAL returns tables with None sycl queue for CPU sycl queue inputs.
+                # This workaround is necessary for the functional preservation
+                # of the compute-follows-data execution.
+                # Host tables first converted into numpy.narrays and then to array from xp
+                # namespace.
+                return xp.asarray(
```

Comment:
This is shouldn't be a follow up ticket of this PR. I don't think that without any input it is needed to create any ticket.
Potentially this could be raised on DBSCAN via Array API enabling with this PR changes integration, since stock scikit-learn doesn't support non-numpy array inputs.